psw/enclave_common/sgx_enclave_common.cpp

```diff
@@ -788,24 +778,9 @@ extern "C" void* COMM_API enclave_create_ex(
         s_enclave_mem_region[enclave_base].prot = 0;
         if(enclave_elrange != NULL)
         {
-            enclave_elrange_t *tmp_enclave_elrange = new(std::nothrow) enclave_elrange_t;
-            if (tmp_enclave_elrange == NULL)
-            {
-                if (enclave_error)
-                {
-                    *enclave_error = ENCLAVE_OUT_OF_MEMORY;
-                }
-
-                //if in-kernel driver then close the file handle
-                if (s_driver_type == SGX_DRIVER_IN_KERNEL)
-                {
-                    close_file(&hdevice_temp);
-                }
-                munmap(enclave_base, virtual_size);
-                return NULL;
-            }
-            memset(tmp_enclave_elrange, 0, sizeof(enclave_elrange_t));
-            if (memcpy_s(tmp_enclave_elrange, sizeof(enclave_elrange_t), enclave_elrange, sizeof(enclave_elrange_t)))
+            enclave_elrange_t tmp_enclave_elrange;
```

Comment:
Like below, you can just directly write into the map. memcpy_s will never fail on these values, so you're not going to be left with a meaningless key.